src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -988,4 +996,20 @@ public static Path getPath(String uriString) throws IOException {
             return FileSystems.newFileSystem(uri, new HashMap<>(), cl).provider().getPath(uri);
         }
     }
+
+    /**
+     * Test whether a input stream looks like a GZIP input.
+     * @param bufferedinput the input stream. The buffer must be large enough to contain {@link GZIP_SIGNATURE}
+     * @return true if `bufferedinput` starts with a gzip signature 
+     * @throws IOException if `bufferedinput` cannot mark or reset the stream
+     * @see {@linkplain http://stackoverflow.com/questions/4818468/how-to-check-if-inputstream-is-gzipped}
+     */
+    public static boolean isGZIPInputStream(final BufferedInputStream bufferedinput) throws IOException {
```

Comment:
Should this be just a simple `InputStream` but check markSupported and then throw if it isn't?  In any case it should probably explicitly check `markSupported()` and throw a helpful error message if it isn't.

Reply:
I noticed there's a similar method `SamStreams.isGzippedSAMFile`. Instead of just checking the header it actually tries to decompress the file and catches exceptions.  Your way may be slightly more elegant, but seems a little more brittle.  In either case, it should probably be refactored to use this method if that's possible.